go/interpreter/lfvm/instructions_test.go

```diff
@@ -1827,6 +1827,48 @@ func TestOpBlockhash(t *testing.T) {
 	}
 }
 
+func TestCallDataload(t *testing.T) {
+
+	zero := uint256.NewInt(0)
+	someData := make([]byte, 32)
+	_, _ = rand.Read(someData)
+	firstByte := someData[31]
+
+	tests := map[string]struct {
+		offset *uint256.Int
+		want   *uint256.Int
+	}{
+		"writes zero on the stack when offset larger than uint64 ": {
+			offset: new(uint256.Int).Add(uint256.NewInt(2), uint256.NewInt(math.MaxUint64)),
+			want:   zero,
+		},
+		"writes zero on the stack when offset+31 overflows uint64": {
+			offset: uint256.NewInt(math.MaxUint64 - 31),
+			want:   zero,
+		},
+		"pads with zeros when offset bigger than input code size": {
+			offset: uint256.NewInt(31),
+			want:   uint256.NewInt(0).SetBytes32([]byte{firstByte, 31: 0x0}),
```

Comment:
nit: since `firstByte` is not used anywhere else, and the name is ambiguous (from the point of view of the input data it is actually the last byte), I'd just inline it here and say
```suggestion
			want:   uint256.NewInt(0).SetBytes32([]byte{someData[31], 31: 0x0}),
```